src/main/java/com/extendedclip/deluxemenus/action/ClickActionTask.java

```diff
@@ -108,20 +112,30 @@ public void run() {
 
       case OPEN_GUI_MENU:
       case OPEN_MENU:
-        final Menu menuToOpen = Menu.getMenu(executable);
+        String[] partsArgs = executable.split(" ");
+        String menuName = partsArgs[0];
+        final Menu menuToOpen = Menu.getMenu(menuName);
         if (menuToOpen == null) {
-          DeluxeMenus.debug(DebugLevel.HIGHEST, Level.WARNING, "Could not find and open menu " + executable);
+          DeluxeMenus.debug(DebugLevel.HIGHEST, Level.WARNING, "Could not find and open menu " + menuName);
           break;
         }
 
-        if (holder == null) {
-          menuToOpen.openMenu(player);
-          break;
+        Map<String, String> argMap = new HashMap<>();
+
+        if (holder != null && holder.getTypedArgs() != null) {
+          argMap.putAll(holder.getTypedArgs());
         }
 
-        menuToOpen.openMenu(player, holder.getTypedArgs(), holder.getPlaceholderPlayer());
+        List<String> expectedArgKeys = menuToOpen.getArgs();
+        for (int i = 0; i < expectedArgKeys.size(); i++) {
+          String value = (i + 1 < partsArgs.length) ? partsArgs[i + 1] : "";
+          argMap.put(expectedArgKeys.get(i), value);
+        }
+
+        menuToOpen.openMenu(player, argMap, holder != null ? holder.getPlaceholderPlayer() : null);

```

Comment:
The idea aligns with what I have in mind but it seems to be somewhat improperly implemented.
- it isn't checked if the new menu has arguments
- all arguments are forced in `[openguimenu]` action (there is no case handling for when there are only some or no arguments specified